benchadapt/python/benchadapt/machine_info.py

```diff
@@ -77,6 +77,31 @@ class GitParseWarning(RuntimeWarning):
 
 
 def github_info():
+    """Get github metadata from environment variables"""
+    repo = os.environ.get("BENCHMARKABLE_REPOSITORY")
+    pr_number = os.environ.get("BENCHMARKABLE_PR_NUMBER")
+    commit = os.environ.get("BENCHMARKABLE_COMMIT")
```

Comment:
Would it make sense to prefix these with `CONBENCH_` so that it's clear that they belong to the conbench (family)?

Reply:
I thought (@austin3dickey ?) we were using `BENCHMARKABLE_PR_NUMBER` somewhere already ([here?](https://github.com/voltrondata-labs/arrow-benchmarks-ci/blob/e4de700016e0d473d82199733b8e4dbe89112b88/models/run.py#L115) but that's internal to arrow-benchmarks-ci, if I understand what's happening there correctly) so the `BENCHMARKABLE_*` prefix was for consistency there, but maybe it's fine if we just make it `CONBENCH_*` everywhere? That would be preferable.

Reply:
Ah I didn't realize we had these already somewhere 😬. So maybe we accept both (for that one, temporarily to make our rollout easier? Or we could set both in the CI for a little bit too? either way is fine...)

But yeah, we probably should have done `CONBENCH_...` from the beginning (oops!). But IMO we aren't _too far_ down the envvarification that catching this now is a good time

Reply:
Yeah, I'm not sure exactly how for the usage of `BENCHMARKABLE_PR_NUMBER` spread, but yeah, I like the plan of accepting it as a backup if we need, but preferring `CONBENCH_*`. I'll adjust.

Reply:
changed

Reply:
Confirming what I read here, in my own words:

If an environment variable enters the stack primarily to configure an aspect about Conbench, then it should be prefixed with `CONBENCH_`. Yes! Super important.

If we use env vars that don't do this so far: let's slowly change them. Config pipelines become a madness otherwise.

We might do some dirty cheating here and there where an environment variable that has central meaning elsewhere is _also_ read by Conbench in order to pragmatically react to it. In that case, the CONBENCH prefix is not applicable.

Reply:
>I'm not sure exactly how for the usage of BENCHMARKABLE_PR_NUMBER spread

The word "benchmarkable" is primarily an `arrow-benchmarks-ci` term that means "the project being benchmarked." Notably, it doesn't exist for other projects like Velox. So I think you'll only find it in https://github.com/voltrondata-labs/arrow-benchmarks-ci and the projects it manages, which right now is just https://github.com/voltrondata-labs/benchmarks.

Agreed; I like using the `CONBENCH_` prefix for these. But I also think these could be a little misleading:
- CONBENCH_PR_NUMBER
- CONBENCH_REPOSITORY
- CONBENCH_COMMIT
- CONBENCH_HOST_NAME

since it feels weird to put an Arrow PR number in a variable called "conbench PR number", or the Arrow repository name in "conbench repository". I also might accidentally put something like https://conbench.ursa.dev/ in CONBENCH_HOST_NAME, confusing it with CONBENCH_URL.

So maybe these should be
- CONBENCH_PROJECT_PR_NUMBER
- CONBENCH_PROJECT_REPOSITORY
- CONBENCH_PROJECT_COMMIT
- CONBENCH_MACHINE_NAME
?

Reply:
Yeah, I appreciate these names are a little ambiguous; they're currently based on a `CONBENCH_` prefix and the key of the element in the request JSON schema—mostly. `pr_number`, `repository`, and `commit` are all a part of `github`, but `CONBENCH_GITHUB_REPOSITORY` is certainly no clearer. `CONBENCH_PROJECT_GITHUB_REPOSITORY`? `CONBENCH_PROJECT_REPOSITORY`? Idk.

Also `CONBENCH_HOST_NAME` really ends up as `machine_info.name`, but I'm not sure if `CONBENCH_MACHINE_INFO_NAME` is clearer that this is the name of a computer. `CONBENCH_MACHINE_INFO_HOST_NAME` implies that it's in `machine_info.host_name` though, which it isn't.

tl;dr: naming is still hard 😭

Reply:
Ok, changed to

- CONBENCH_PROJECT_PR_NUMBER
- CONBENCH_PROJECT_REPOSITORY
- CONBENCH_PROJECT_COMMIT
- CONBENCH_MACHINE_INFO_NAME

Don't love them, but they're a balance between specificity and concision.